R/utilities-simulation-results.R

```diff
@@ -77,18 +80,29 @@ getOutputValues <- function(simulationResults,
   # Cache of all individual properties over all individual that will be duplicated in all resulting data.frame
   allIndividualProperties <- do.call(rbind.data.frame, c(individualPropertiesCache, stringsAsFactors = FALSE))
 
-
   values <- lapply(paths, function(path) {
     simulationResults$getValuesByPath(path, individualIds, stopIfNotFound)
   })
   names(values) <- paths
 
-  metaData <- lapply(paths, function(path) {
-    quantity <- quantities[[path]]
-    list(unit = quantity$unit, dimension = quantity$dimension)
-  })
-  names(metaData) <- paths
-  metaData[["Time"]] <- list(unit = "min", dimension = "Time")
+  # Use low-level methods to get unit and dimension
+  task <- getContainerTask()
+  metaData <- NULL
+  if (addMetaData) {
+    metaData <- lapply(paths, function(path) {
+      unit <- NULL
+      dimension <- NULL
+      # Get the dimension and unit from path if the results are obtained. If the results
+      # are NA, the entity with such path does not exist
+      if (!all(is.na(values[[path]]))) {
+        unit <- rClr::clrCall(task, "BaseUnitNamesFromPath", simulationResults$simulation$ref, enc2utf8(path))[[1]]

```

Comment:
I am wondering if this function should not return only values for one path as opposed as for an array.
It looked suspicious to me before but seeing how it's used now...I think the implementation in .NET should only work with one path. Or do you expect to have wild cards?

Reply:
Good point, I have to check what happens when passing a path with wildcards. Will do it tomorrow.

Reply:
According to the description, 

> #' When providing the paths, only absolute full paths are supported (i.e., no matching with '*' possible).

And the called .NET method is `public virtual float[] AllValuesFor(string quantityPath, params int[] individualIds)` of `SimulationResults`, which probably only accepts absolute paths.

Actually when using the wildcard, the path is not transformed to a regexp and the error is 

`getOutputValues: No results exists for path 'Organism|PeripheralVenousBlood|*|Plasma (Peripheral Venous Blood)' for individual IDs '0'! `



So maybe the funciton in .NET should always return only one value.

Reply:
ok. Going to update the .net side now